### DISCUSSION TITLE
XAML - Platform Specific - How to determine a Tablet vs a Phone form factor

### DISCUSSION THREAD

**Discussion by crogeniks** [AvaloniaUI/Avalonia#16508] [Q&A] [answered]
Today, Avalonia provides two ways to create Platform Specific XAML
- `OnPlatform`
- `OnFormFactor`

With `OnPlatform`, I can change my UI depending on the platform, Desktop vs Android vs iOS
With `OnFormFactor`, I can change my UI depending on if the app runs in a Desktop environment or a Mobile environment.

However, `OnFormFactor.Mobile` does not differentiate between a Phone and a Tablet form factor. What would be the best way to achieve this?

**Comment by maxkatz6** [comment]
"Tablet" form factor is a problematic and is very difficult to reliably differentiate from Mobile. Or even Desktop. Is Surface Pro a Tablet? Is android phone with a bigger screen a tablet?

In my opinion, it doesn't make a good responsive app to rely on tablet form factor in this situation. The only definitive attribute is a big enough screen and touch input - so you should use these attributes instead. Screen (or top-level control) size and last input pointer type (last one, because it can be in fact changed, user might switch to mouse, and some apps might want to adjust UI too).

**Reply by crogeniks** [reply]
When I say Tablet vs Phone, I really only mean the screen size, not the actual device underneath.

Can you point me on documentation that can help me achieve this?

Concrete example is a Master-Detail screen. On a "Large" screen I would want a Side panel with a content panel, whereas on a "Small" screen I would hide the content panel.

**Reply by maxkatz6** [reply, marked as answer]
> On a "Large" screen I would want a Side panel with a content panel, whereas on a "Small" screen I would hide the content panel.

Right now, you need to subscribe on SizeChanged event and adjust your styling accordingly.
You could use `Avalonia.Xaml.Behaviors` which has some adaptive extensions: https://github.com/AvaloniaUI/Avalonia.Xaml.Behaviors/blob/d62d2aa68e3943b26b8f33cc120f2e8cc707e038/samples/BehaviorsTestApplication/Views/Pages/AdaptiveBehaviorView.axaml#L18
We were planning to implement feature similar to `@media` selectors in css, but it was postponed. It would be a syntax sugar on top of SizeChanged event essentially.

**Reply by crogeniks** [reply]
`AdaptiveBehavior` looks exactly like what I would need/like to use actually.

Awesome. Thanks for your input.
